src/sparse/impl/KokkosSparse_spmv_struct_impl.hpp

```diff
@@ -507,11 +508,19 @@ struct SPMV_Struct_Functor {
         rowIdx = (k + 1)*nj*ni + (j + 1)*ni + (i + 1);
 
         const size_type rowOffset = m_A.graph.row_map(rowIdx);
+
+	y_value_type sum(0.0);
+#ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST
+        for (ordinal_type idx = 0; idx < 27; ++idx) {
+         sum += m_A.values(rowOffset + idx)*m_x(rowIdx + columnOffsets(idx));
+        }
+#else
         const value_type* value_ptr = &(m_A.values(rowOffset));
-	value_type sum = 0.0;
-	Kokkos::parallel_reduce(Kokkos::ThreadVectorRange(dev, 27), [&] (const ordinal_type& idx, value_type& lclSum) {
-	    lclSum += *(value_ptr + idx)*m_x(rowIdx + columnOffsets(idx));
-	  }, sum);
+	Kokkos::parallel_reduce(Kokkos::ThreadVectorRange(dev, 27), [&] (const ordinal_type& idx, y_value_type& lclSum) {
```

Comment:
lclSum and sum should have always had the type `y_value_type`.  Otherwise, mixed-precision residuals won't work.  Glad it's right now : - )